sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -199,7 +201,12 @@ boolean isContentResponseOnWriteEnabled() {
      * an error.
      */
     public Mono<CosmosAsyncDatabaseResponse> createDatabaseIfNotExists(CosmosDatabaseProperties databaseSettings) {
-        return createDatabaseIfNotExistsInternal(getDatabase(databaseSettings.getId()));
+        if(!getTracerProvider().isEnabled()) {
```

Comment:
The `withContext` overload would be used for context passing and not just tracing context-related so should not be opted in for regardless of tracing presence.
@srnagar